samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor

```diff
@@ -175,93 +128,157 @@
+        PagedNodeList = NodeList.GetPaged(page, 10);
+        foreach (var node in PagedNodeList.Results)
+        {
+            // fetch the actual value
+            if (node.NodeClass == NodeClass.Variable)
+            {
+                node.Value = await BrowseManager.ReadValueAsync(endpointId, node.Id);
+            }
+        }
         NavigationManager.NavigateTo("/browser/" + supervisorId + "/" + endpointId + "/" + page);
     }
 
+    /// <summary>
+    /// OnInitializedAsync
+    /// </summary>
+    /// <returns></returns>
     protected override async Task OnInitializedAsync()
     {
         await BrowseTreeAsync(string.Empty, new List<string>(), BrowseDirection.Forward);
     }
 
+    /// <summary>
+    /// GetTreeAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <returns></returns>
     private async Task GetTreeAsync(string id, List<string> parentId)
     {
         await BrowseTreeAsync(id, parentId, BrowseDirection.Forward);
     }
 
+    /// <summary>
+    /// GetTreeBackAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <returns></returns>
     private async Task GetTreeBackAsync(string id, List<string> parentId)
     {
         await BrowseTreeAsync(id, parentId, BrowseDirection.Backward);
         NavigationManager.NavigateTo("/browser/" + supervisorId + "/" + endpointId);
     }
 
+    /// <summary>
+    /// BrowseTreeAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <param name="direction"></param>
+    /// <returns></returns>
     private async Task BrowseTreeAsync(String id, List<string> parentId, BrowseDirection direction)
     {
         NodeList = await BrowseManager.GetTreeAsync(endpointId, id, parentId, supervisorId, direction);
-        Page = "1";
-        pagedNodeList = NodeList.GetPaged(Int32.Parse(Page), 10);
-    }
+        var publishedNodes = await Publisher.PublishedAsync(endpointId);
 
-    private void OpenDrawer(ListNode node)
-    {
-        IsOpened = true;
-        nodeData = node;
-    }
+        foreach (var node in NodeList.Results)
+        {
+            // fetch the actual value
+            if (node.NodeClass == NodeClass.Variable)
+            {
+                // check if publishing enabled
+                foreach (var publishedNode in publishedNodes.Results)
+                {
+                    if (node.Id == publishedNode.NodeId)
+                    {
+                        node.PublishedNode = publishedNode;
+                        node.Publishing = true;
+                        break;
+                    }
+                }
+            }
+        }
 
-    private void CloseDrawer()
-    {
-        IsOpened = false;
-        TypeOfAction = ActionType.Nothing;
+        Page = "1";
+        PagedNodeList = NodeList.GetPaged(Int32.Parse(Page), 10);
```

Comment:
>10 [](start = 61, length = 2)

is this pagesize? If yes, have at least a global const to be able to change easily lateron.

Reply:
added const value